pkg/apimachinery/validation/validation.go

```diff
@@ -0,0 +1,105 @@
+package validation
+
+import (
+	"regexp"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+const maxNameLength = 253
+const maxNamespaceLength = 40
+const minNamespaceLength = 3
+const maxGroupLength = 60
+const minGroupLength = 3
+const maxResourceLength = 40
+const minResourceLength = 3
+
+const grafanaNameFmt = `^[a-zA-Z0-9:\-\_\.]*$`
+const grafanaNameErrMsg string = "must consist of alphanumeric characters, '-', '_', ':' or '.'"
+
+const qnameCharFmt string = "[A-Za-z0-9]"
+const qnameExtCharFmt string = "[-A-Za-z0-9_.]"
+const qualifiedNameFmt string = "^(" + qnameCharFmt + qnameExtCharFmt + "*)?" + qnameCharFmt + "$"
+const qualifiedNameErrMsg string = "must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character"
+
+const alphaCharFmt string = "[A-Za-z]"
+const resourceCharFmt string = "[-A-Za-z0-9_]" // no "." allowed
+const resourceFmt string = "^" + alphaCharFmt + resourceCharFmt + "*$"
+const resourceErrMsg string = "must consist of alphanumeric characters or '_', and must start and end with an alphanumeric character"
+
+var (
+	grafanaNameRegexp   = regexp.MustCompile(grafanaNameFmt).MatchString
+	qualifiedNameRegexp = regexp.MustCompile(qualifiedNameFmt).MatchString
+	resourceRegexp      = regexp.MustCompile(resourceFmt).MatchString
+)
+
+// IsValidGrafanaName checks if the name is a valid to use for a k8s name
+// Unlike normal k8s name rules, this allows the name to start with a digit
+// This compromise means existing grafana UIDs are valid k8s names without migration
+func IsValidGrafanaName(name string) []string {
+	s := len(name)
+	switch {
+	case s == 0:
+		return []string{"name may not be empty"}
+	case s > maxNameLength:
```

Comment:
do we have a minimum grafana name length?

would a name like `x` be a valid grafana name?

Reply:
we do not have one now, so I am nervous to add it -- since it is already possible to set them with legacy APIs that we want to migrate.

Reply:
so yes "x" would be valid grafana UID or k8s name